Project with PackAsTool and GeneratePackageOnBuild set to true fails to build

Tests in `GivenThatWeWantToPackAToolProjectWithGeneratePackageOnBuild` are failing with:

> C:\git\dotnet-sdk\artifacts\bin\Debug\Sdks\NuGet.Build.Tasks.Pack\build\NuGet.Build.Tasks.Pack.targets(337,5): error MSB4006: There is a circular dependency in the target dependency graph involving target "_WalkEachTargetPerFramework". [C:\git\dotnet-sdk\artifacts\tmp\Debug\It_builds_suc---81F3C59C\App\consoledemo.csproj]

This regressed with a recent NuGet insertion.

--- a/src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Publish.targets
+++ b/src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Publish.targets
@@ -53,11 +53,11 @@ Copyright (c) .NET Foundation. All rights reserved.
   </PropertyGroup>
 
   <Target Name="_PublishBuildAlternative"
-          Condition="'$(NoBuild)' != 'true'"
+          Condition="'$(NoBuild)' != 'true' and '$(GeneratePackageOnBuild)' != 'true'"
           DependsOnTargets="Build;$(_CorePublishTargets)" />
 
   <Target Name="_PublishNoBuildAlternative"
-          Condition="'$(NoBuild)' == 'true'"
+          Condition="'$(NoBuild)' == 'true' or '$(GeneratePackageOnBuild)' == 'true'"
           DependsOnTargets="$(_BeforePublishNoBuildTargets);$(_CorePublishTargets)" />
 
   <Target Name="Publish"

--- a/src/Tests/Microsoft.NET.ToolPack.Tests/GivenThatWeWantToPackAToolProjectWithGeneratePackageOnBuild.cs
+++ b/src/Tests/Microsoft.NET.ToolPack.Tests/GivenThatWeWantToPackAToolProjectWithGeneratePackageOnBuild.cs
@@ -46,7 +46,7 @@ namespace Microsoft.NET.ToolPack.Tests
             return testAsset;
         }
 
-        [Fact(Skip = "https://github.com/dotnet/sdk/issues/3253")]
+        [Fact]
         public void It_builds_successfully()
         {
             TestAsset testAsset = SetupAndRestoreTestAsset();
@@ -61,7 +61,7 @@ namespace Microsoft.NET.ToolPack.Tests
                   .NotHaveStdOutContaining("There is a circular dependency");
         }
 
-        [Fact(Skip = "https://github.com/dotnet/sdk/issues/3253")]
+        [Fact]
         public void It_builds_and_result_contains_dependencies_dll()
         {
             TestAsset testAsset = SetupAndRestoreTestAsset();

